Bottom: a5357b671a680a91831aa153a0ddd5db2e156cd8
Top:    5728eb829f8905ba753c4bb8444a3b98d250536e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 20:01:08 +0000

OLD HOLD-OVER, THIS FIX WAS LIKELY INCORRECT, AND NOW REDUNDANT. PRESERVING FOR POSTERITY FOR NOW...

polygon1.c: Fix Collect routine

Not 100% sure about this, but it seems to fix a nasty
case regarding incorrect geometry created in the following
test geometry:

Element["" "" "" "unknown" 1325.00mil 1070.00mil -175.00mil -105.00mil 0 100 ""]
(
  Pad[-85.00mil -39.00mil -79.00mil -39.00mil 34.00mil 30.00mil 40.00mil "2" "2" "square"]
)

Layer(1 "component")
(
  Line[1245.00mil 1030.00mil 1195.00mil 1030.00mil 12.00mil 20.00mil "clearline"]
  Polygon("clearpoly")
  (
    [5.00mil 5.00mil] [4195.00mil 5.00mil] [4195.00mil 2695.00mil] [5.00mil 2695.00mil]
  )
)


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 460ad55..2942c3b 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1723,6 +1723,10 @@ Collect (char poly, jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes
         Collect1 (e, (dir == FORW) ? EDGE_BACKWARD_VERTEX (cure) :
                                      EDGE_FORWARD_VERTEX (cure),
                   dir, contours, holes, j_rule);
+//    The lines below are no-longer required.. preserving for now to record this commit.
+//      other = cur;
+//      if (other->cvc_prev)
+//	Collect1 (e, other, dir, contours, holes, j_rule);
     }
   while ((cure = NEXT_EDGE (cure)) != &a->head);
 }				/* Collect */
